plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -26,17 +27,364 @@ export interface ProductCollectionConfig extends BlockVariation {
 	};
 }
 
+const isValidProductCollectionConfig = ( config: ProductCollectionConfig ) => {
+	// Basic checks for the top-level argument
+	if ( typeof config !== 'object' || config === null ) {
+		console.error(
+			'Invalid arguments: You must pass an object to __experimentalRegisterProductCollection.'
+		);
+		return false;
+	}
+
+	// BlockVariation properties
+	if ( typeof config.name !== 'string' || config.name.length === 0 ) {
+		console.error( 'Invalid name: name must be a non-empty string.' );
+		return false;
+	}
+	if (
+		config.title !== undefined &&
+		( typeof config.title !== 'string' || config.title.length === 0 )
+	) {
+		console.error( 'Invalid title: title must be a non-empty string.' );
+		return false;
+	}
+	if (
+		config.description !== undefined &&
+		typeof config.description !== 'string'
+	) {
+		console.error( 'Invalid description: description must be a string.' );
+		return false;
+	}
+	if (
+		config.category !== undefined &&
+		typeof config.category !== 'string'
+	) {
+		console.error( 'Invalid category: category must be a string.' );
+		return false;
+	}
+	if ( config.keywords !== undefined && ! Array.isArray( config.keywords ) ) {
+		console.error(
+			'Invalid keywords: keywords must be an array of strings.'
+		);
+		return false;
+	}
+	if (
+		config.icon !== undefined &&
+		typeof config.icon !== 'string' &&
+		typeof config.icon !== 'object'
+	) {
+		console.error( 'Invalid icon: icon must be a string or an object.' );
+		return false;
+	}
+	if (
+		config.isDefault !== undefined &&
+		typeof config.isDefault !== 'boolean'
+	) {
+		console.error( 'Invalid isDefault: isDefault must be a boolean.' );
+		return false;
+	}
+	if (
+		config.innerBlocks !== undefined &&
+		! Array.isArray( config.innerBlocks )
+	) {
+		console.error( 'Invalid innerBlocks: innerBlocks must be an array.' );
+		return false;
+	}
+	if ( config.example !== undefined && typeof config.example !== 'object' ) {
+		console.error( 'Invalid example: example must be an object.' );
+		return false;
+	}
+	if ( config.scope !== undefined && ! Array.isArray( config.scope ) ) {
+		console.error(
+			'Invalid scope: scope must be an array of type WPBlockVariationScope.'
+		);
+		return false;
+	}
+	if (
+		config.isActive !== undefined &&
+		typeof config.isActive !== 'function'
+	) {
+		console.error( 'Invalid isActive: isActive must be a function.' );
```

Comment:
We agreed `isActive` is not a subject to change:

> You can see all other supported arguments in the [Block Variation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/#defining-a-block-variation) documentation. You won't be able to use isDefault attribute as we are setting it to false for all 3PDs collections. Also, isActive can't be used as it will be managed by us.

from https://github.com/woocommerce/woocommerce/pull/48141 description, hence property should be ignored completely.